server/src/main/java/org/opensearch/action/bulk/BulkItemResponse.java

```diff
@@ -94,6 +94,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.field(_ID, failure.getId());
             builder.field(STATUS, failure.getStatus().getStatus());
             builder.startObject(ERROR);
+            builder.field(_INDEX, failure.getIndex());
+            builder.field(_ID, failure.getId());
```

Comment:
Is this always generated when the error is passed? What if the error was encountered even before the document id could be generated?

Reply:
Then the behaviour would be the same as in for `builder.field(_ID, failure.getId());` that is above line `builder.startObject(ERROR);`